addons/web_editor/controllers/main.py

```diff
@@ -182,6 +183,13 @@ def _update_checklist_recursive (self, li, checked, children=False, ancestors=Fa
 
         return True
 
+    @http.route('/web_editor/attachment/fetch', type='json', auth='user', methods=['POST'], website=True)
+    def fetch_attachments(self, domain=None, fields=None, offset=0, limit=None, order=None, **read_kwargs):
+        if request.env.user.has_group('base.group_portal'):
+            domain = expression.AND([[['create_uid', '=', request.env.uid]], domain])
+            return request.env['ir.attachment'].sudo().search_read(domain, fields, offset, limit, order, **read_kwargs)
```

Comment:
IMO if this was done, this should rather be:

```py
if not self.env.user.has_group('base.group_user'): # and possibly 'and request.env.user.has_group('base.group_portal')'
    expression.AND([[['create_uid', '=', request.env.uid], ['public', '=', True]], domain])
    return request.env['ir.attachment'].sudo().search_read(domain, fields, offset, limit, order, **read_kwargs)
```

currently this allows portal user to read all attachments where create_uid==request.env.uid (which is not something that is allowed even for an admin user)

but it might be enough to just `return []`